simple_graph.py

```diff
@@ -8,65 +8,72 @@ def nodes(self):
         return self.graph.keys()
 
     def edges(self):
-        """Return a list of edges in the graph."""
+        """Return a list of edges and their weights in the graph."""
         l = []
         for k, v in self.graph.iteritems():
             for node in v:
-                l.append([k, node])
+                l.append([k, node, v[node]])
         return l
 
     def add_node(self, node):
         """Add a node to the graph."""
-        self.graph.setdefault(node, [])
+        self.graph.setdefault(node, {})
 
-    def add_edge(self, node1, node2):
+    def add_edge(self, node1, node2, weight):
```

Comment:
I still think it'd be better to leave the external api of this method unaltered by making `weight` an optional argument